client/network/src/protocol.rs

```diff
@@ -528,10 +528,9 @@ impl<B: BlockT, H: ExHashT> Protocol<B, H> {
 		self.sync.num_sync_requests()
 	}
 
-	/// Sync local state with the blockchain state.
-	pub fn update_chain(&mut self) {
-		let info = self.context_data.chain.info();
-		self.sync.update_chain_info(&info.best_hash, info.best_number);
+	/// Inform sync about new best imported block.
+	pub fn new_best_block_imported(&mut self, hash: B::Hash, number: NumberFor<B>) {
+		self.sync.update_chain_info(&hash, number);
```

Comment:
Should probably also call `self.behaviour.set_notif_protocol_handshake` here, @tomaka?

Reply:
I'm doing it. Just look down :D

Reply:
I think it's called right below, just not shown in the diff.